### PR TITLE
Add staff submission view and submission delete and unsubmit buttons

### PR DESCRIPTION
**Pull request by ekowidianto**

### CODE REVIEW

**Comment by ekowidianto** [comment]
Codecov is low as test in CircleCI is flaky. submissions_spec and download_spec tests work locally

**Comment by zhuhanming** [comment]
To confirm - are we no longer doing the checkbox thing? Where users can select the submissions they want to download/unsubmit/etc., then do a mass operation.

Also, I think the tag UI doesn't look good for the phantom students. Let me think about this a bit more.

**Comment by ekowidianto** [comment]
> To confirm - are we no longer doing the checkbox thing? Where users can select the submissions they want to download/unsubmit/etc., then do a mass operation.
> 
> Also, I think the tag UI doesn't look good for the phantom students. Let me think about this a bit more.

Yup, as of now we stick to the current UI design.

**Comment by zhuhanming** [comment]
I feel that this breakdown is not ideal. Can we instead do **Students, Staff, Phantom**? (and if I'm a staff in a group, My Students at the front as well)

Reasoning is as such (written from the perspective of a course coordinator):

- If we lump the phantom people with the respective groups, then they will end up in my statistics, exports, mass operations etc. - which is not desirable. Generally, I only care about the actual students and staff in my course. Especially in terms of exporting their data - after exporting the data, I lose track of who's phantom and who's not, and will need to cross check with the list on Coursemology. Adding a new column in the exported CSV is also not very ideal - very messy. I also can't really do much for downloading submissions, e.g. for programming assignments, since what's downloaded is not a CSV, but folders of code.
   - The number of phantom people generally is limited, and I generally know who are they, since I've added them separately, potentially for differing reasons. As such, it's fine that they get excluded from certain mass operations (e.g. phantom students being excluded from operations done on all students) - I can go through the phantom list and perform these operations myself if necessary, even if I need to do so one by one.
- As a course coordinator, I don't really care if the phantom is a staff or a student - we don't actually need that granularity. If you just lump them together under a Phantom category, it's fine by me.

TL;DR - there is **quite a lot more pain** caused from lumping the phantom with the normal, than the benefits I'll gain from doing so. Also, the chip UI makes the UI very cluttered and confusing (and also quite unaesthetic) - this change will allow us to avoid having to use the chip.

cc: @cysjonathan

**Comment by ekowidianto** [comment]
> I feel that this breakdown is not ideal. Can we instead do **Students, Staff, Phantom**? (and if I'm a staff in a group, My Students at the front as well)
> 
> Reasoning is as such (written from the perspective of a course coordinator):
> 
>     * If we lump the phantom people with the respective groups, then they will end up in my statistics, exports, mass operations etc. - which is not desirable. Generally, I only care about the actual students and staff in my course. Especially in terms of exporting their data - after exporting the data, I lose track of who's phantom and who's not, and will need to cross check with the list on Coursemology. Adding a new column in the exported CSV is also not very ideal - very messy. I also can't really do much for downloading submissions, e.g. for programming assignments, since what's downloaded is not a CSV, but folders of code.
>       
>       * The number of phantom people generally is limited, and I generally know who are they, since I've added them separately, potentially for differing reasons. As such, it's fine that they get excluded from certain mass operations (e.g. phantom students being excluded from operations done on all students) - I can go through the phantom list and perform these operations myself if necessary, even if I need to do so one by one.
> 
>     * As a course coordinator, I don't really care if the phantom is a staff or a student - we don't actually need that granularity. If you just lump them together under a Phantom category, it's fine by me.
> 
> 
> TL;DR - there is **quite a lot more pain** caused from lumping the phantom with the normal, than the benefits I'll gain from doing so. Also, the chip UI makes the UI very cluttered and confusing (and also quite unaesthetic) - this change will allow us to avoid having to use the chip.
> 
> cc: @cysjonathan

The initial consideration to lump phantom people together:
- There is phantom staff as well and in the end there will be 5 tabs including my students 
- There is this phantom student toggle on top. Users can be misled into thinking that if it is toggled on, phantom users are included in the list
- Also, there is currently no way to tell if a student is a phantom student in the my student column

To address your concern:
- "....then they will end up in my statistics, exports, mass operations etc...." -> that's where the phantom toggle comes in. If it is not turned on, then an instructor is just simply performing mass operations on the students in the list (normal users). 

In general, I am fine to revert to the previous UI design which then means the only changes made would be only adding a staff tab and unsubmission/delete buttons

cc @cysjonathan what do you think?

**Comment by cysjonathan** [comment]
For chip display, how about we do it similar to assessments list page when we label assessments as personal timeline
image

We can label users as "phantom" with an icon+tooltip instead
https://fontawesome.com/v5.15/icons/ghost?style=solid

I think we should minimize the number of tabs.
Phantoms are toggled off by default on the initial page view.
If users are looking for phantoms, they will naturally gravitate to the phantom toggle and make the phantom users (students/staff) visible.

**Comment by zhuhanming** [comment]
> For chip display, how about we do it similar to assessments list page when we label assessments as personal timeline image
> 
> We can label users as "phantom" with an icon+tooltip instead https://fontawesome.com/v5.15/icons/ghost?style=solid
> 
> I think we should minimize the number of tabs. Phantoms are toggled off by default on the initial page view. If users are looking for phantoms, they will naturally gravitate to the phantom toggle and make the phantom users (students/staff) visible.

Ok this sounds good to me. Agreed on disabling the phantom toggle by default.

Should we have some indication of phantomness in the exported e.g. CSV files? Or not necessary?

**Comment by ekowidianto** [comment]
> > For chip display, how about we do it similar to assessments list page when we label assessments as personal timeline image
> > We can label users as "phantom" with an icon+tooltip instead https://fontawesome.com/v5.15/icons/ghost?style=solid
> > I think we should minimize the number of tabs. Phantoms are toggled off by default on the initial page view. If users are looking for phantoms, they will naturally gravitate to the phantom toggle and make the phantom users (students/staff) visible.
> 
> Ok this sounds good to me. Agreed on disabling the phantom toggle by default.
> 
> Should we have some indication of phantomness in the exported e.g. CSV files? Or not necessary?

In fact, i have already added an additional column in the csv file indicating the phantomness

**Comment by zhuhanming** [comment]
> In fact, i have already added an additional column in the csv file indicating the phantomness

I see - in that case, shall we just implement the UI change the Jon suggested?

**Comment by ekowidianto** [comment]
> > In fact, i have already added an additional column in the csv file indicating the phantomness
> 
> I see - in that case, shall we just implement the UI change the Jon suggested?

Would look something like this if thats ok!

image

**Comment by zhuhanming** [comment]
Can make the icon the same size and colour as the ones for assessments, i.e. those in the screenshot Jonathan shared?

Also, is there a more "semantic" icon? I don't see this person icon as something that represents phantomness.

I see the following as potential candidates (from MUI icons):

- PrivacyTip
- RemoveCircle
- Block (kinda like Assessments)

Personally, I liked PrivacyTip quite a lot, but Block is also ok I think.

**Comment by zhuhanming** [comment]
Also the icon should not be part of the hyperlink. It should be before the hyperlink, with some space after it.

**Comment by ekowidianto** [comment]
> Can make the icon the same size and colour as the ones for assessments, i.e. those in the screenshot Jonathan shared?
> 
> Also, is there a more "semantic" icon? I don't see this person icon as something that represents phantomness.
> 
> I see the following as potential candidates (from MUI icons):
> 
>     * PrivacyTip
> 
>     * RemoveCircle
> 
>     * Block (kinda like Assessments)
> 
> 
> Personally, I liked PrivacyTip quite a lot, but Block is also ok I think.

Found a better one

https://fontawesome.com/v4.7/icon/user-secret

**Comment by ekowidianto** [comment]
Codecov is low as test in tests dont pass in CircleCI, although submissions_spec and download_spec tests work locally. Consider making these tests pending?

Otherwise, ready for review
